include/seqan3/alignment/all.hpp

```diff
@@ -10,20 +10,293 @@
+ *<td style="background: #90ff90; color: black; vertical-align: middle; text-align: center; border: 1px solid black;" class="table-yes"> ✓ </td>
+ *<td style="background: #90ff90; color: black; vertical-align: middle; text-align: center; border: 1px solid black;" class="table-yes"> ✓ </td>
+ *<td style="background: #90ff90; color: black; vertical-align: middle; text-align: center; border: 1px solid black;" class="table-yes"> ✓ </td>
+ *<td style="background: #ff9090; color: black; vertical-align: middle; text-align: center; border: 1px solid black;" class="table-no"> ✘ </td>
+ *</tr>
+ *</table>
+ *
+ * ## Accessing the computed alignment
+ *
+ * The seqan3::align_pairwise interface returns a seqan3::alignment_range. This range is a single pass range over
+ * the computed alignments and dereferencing the iterator over this range will return seqan3::alignment_result objects.
+ * Even if only a single alignment is computed a range will be returned since it could be possible that
+ * one alignment invocation produces multiple results, e.g. to receive all suboptimal alignments.
+ * The seqan3::alignment_result object contains only the information that have been requested by the alignment
+ * configuration. The seqan3::align_cfg::result configuration element allows to limit the parts that are computed
+ * in the alignment algorithm. The following table shows the parts that are computed depending on the
+ * seqan3::align_cfg::result configuration:
+ *
+ * | **Entity**                                                                     | **Available result**                                  |
+ * | -------------------------------------------------------------------------------|------------------------------------------------------ |
+ * | \ref seqan3::align_cfg::result::with_score  "with_score"                       | alignment score                                       |
+ * | \ref seqan3::align_cfg::result::with_back_coordinate  "with_back_coordinate"   | alignment score; back coordinate                      |
+ * | \ref seqan3::align_cfg::result::with_front_coordinate  "with_front_coordinate" | alignment score; back and front coordinate            |
+ * | \ref seqan3::align_cfg::result::with_alignment  "with_alignment"               | alignment score; back and front coordinate; alignment |
  *
- * The alignment module contains concepts, functions and classes that are related to the computation of
- * pairwise and multiple sequence alignments.
+ * The back coordinate stores the end of the alignment within both sequences. Note that theses positions are
+ * inclusive. The front coordinate stores the begin of the alignment part in both sequences.
+ * With the last option the complete alignment is computed. If seqan3::align_cfg::result is not specified only the
+ * score is available. Accessing one of the results that have not been requested will trigger a static assertion
+ * informing the developer about the invalid access.
```

Comment:
Either here or in the align_cfg::result documentation a snippet for these `with_*` elements would be helpful.

Reply:
it is in the cfg documentation itself